### ISSUE TITLE
meta: new ZeroMQ/C++ license

### ISSUE THREAD

**Issue by sbinet** [go-zeromq/zmq4#145]
with the ZeroMQ C++ project being relicensed under MPL-2:

https://github.com/zeromq/libzmq/pull/4555

we could theoretically rewrite go-zeromq/zmq4 by "just" reading ZeroMQ/C++ code.
(that's still a lot of work, but we could almost get into a bug-for-bug state)

**Comment by ahmetson** [comment]
Yeah let’s do it. But I will change the API interface to match with pebbe/zmq4.

since that is already released, I think there might be more production libraries relying on library. It will be easier to migrate.

**Comment by ahmetson** [comment]
@sbinet can we open a telegram or some other library? I will write it from scratch.

**Comment by sbinet** [comment]
@ahmetson if you want to tackle this (great!), I'd advise just to create a branch off the latest `go-zeromq/zmq4` `main` branch, remove all (code) files and add your translated from C++ files.

also, it would be great to keep the Go/CGo compatibility tests :)
(I've updated this issue to add the list of ZMQ sockets to implement)

**Comment by ahmetson** [comment]
@sbinet, how is the CGO compatible with the rewritten zmq? 

To make the compatibility, then the current interface must be kept. So, no pebbe/zmq4?

**Comment by sbinet** [comment]
not necessarily.
what I meant was to keep **a** set of Go/CGo compatibility tests, but we can change the interface to be more in line with `pebbe/zmq4`'s.

**Comment by ahmetson** [comment]
Okay, I will start in December. Until 10 December, I will be in the hackathon. I will fork this repo and start slowly implementing them.
As I start, I will keep you updated.

**Comment by sbinet** [comment]
Cool, sounds like a plan.
Keep us posted.
